065-topic-replication-factor-change.md

```diff
@@ -0,0 +1,308 @@
+{
+  "summary": {
+    "numIntraBrokerReplicaMovements": 0,
+    "numReplicaMovements": 125,
+    "onDemandBalancednessScoreAfter": 100,
+    "intraBrokerDataToMoveMB": 0,
+    "monitoredPartitionsPercentage": 100,
+    "provisionRecommendation": "",
+    "excludedBrokersForReplicaMove": [],
+    "excludedBrokersForLeadership": [],
+    "provisionStatus": "RIGHT_SIZED",
+    "onDemandBalancednessScoreBefore": 100,
+    "recentWindows": 5,
+    "dataToMoveMB": 0,
+    "excludedTopics": [],
+    "numLeaderMovements": 0
+  }
+  ...
+```
+
+The `skip_rack_awareness_check` parameter configures whether to skip the rack awareness sanity check or not (default to false).
+If rack awareness is not enabled in Strimzi, the `skip_rack_awareness_check` will be set to true.
+
+The `goals` parameter is a comma separated list of goals used to generate the automatic cluster rebalance for the replication factor change.
+This is configurable for all endpoints in `Kafka.spec.cruiseControl.config` using the `default.goals` property.
+
+The `replication_throttle` parameter is an upper bound on the bandwidth used to move replicas (bytes per second).
+This is configurable for all endpoints in `Kafka.spec.cruiseControl.config` using the `default.replication.throttle` property.
+
+### User tasks
+
+The [`user_tasks`](https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-user-request-result) endpoint will be used to periodically check the state of requested tasks.
+The only required parameter is the `user_task_ids`, which is a comma separated list of `User-Task-ID` returned by the topic configuration requests.
+
+```sh
+$ curl -s "localhost:9090/kafkacruisecontrol/user_tasks?user_task_ids=5344ca89-351f-4565-8d0f-9aade00e053d,8911ca89-351f-888-8d0f-9aade00e098h&json=true" | jq
+{
+  "userTasks": [
+    {
+      "Status": "Completed",
+      "ClientIdentity": "127.0.0.1",
+      "RequestURL": "POST /kafkacruisecontrol/topic_configuration?topic=my-topic&replication_factor=2&skip_rack_awareness_check=true&dryrun=false&json=true",
+      "UserTaskId": "5344ca89-351f-4565-8d0f-9aade00e053d",
+      "StartMs": "1701961135978"
+    },
+    {
+      "Status": "Active",
+      "ClientIdentity": "127.0.0.1",
+      "RequestURL": "POST /kafkacruisecontrol/topic_configuration?topic=my-topic&replication_factor=3&skip_rack_awareness_check=true&dryrun=false&json=true",
+      "UserTaskId": "8911ca89-351f-888-8d0f-9aade00e098h",
+      "StartMs": "1701961136014"
+    }
+  ],
+  "version": 1
+}
+```
+
+### Configuration parameters
+
+The Topic Operator will need to authenticate and send HTTP requests to the Cruise Control's REST API.
+A new admin user will be provisioned for the Topic Operator, separate from the credentials used by the KafkaRebalanceOperator.
+The following environment variables will be set by the Cluster Operator for consumption by the Topic Operator when Cruise Control is configured in Kafka resource, or manually by the user in case of standalone deployment:
```

Comment:
```suggestion
The following environment variables will be set by the Cluster Operator for consumption by the Topic Operator when Cruise Control is configured in the Kafka resource, or manually by the user in case of standalone deployment:
```